contrib/kae/BUILD

```diff
@@ -0,0 +1,33 @@
+load("@rules_foreign_cc//foreign_cc:configure.bzl", "configure_make")
+load("//bazel:envoy_build_system.bzl", "envoy_contrib_package")
+load(
+    "//contrib:all_contrib_extensions.bzl",
+    "envoy_contrib_linux_aarch64_constraints",
+)
+
+licenses(["notice"])  # Apache 2
+
+envoy_contrib_package()
+
+# Have uadk (KAE driver) outside of any extensions because it may be used by both
+# KAE compression library extension and KAE private key provider
+# extension.
+
+configure_make(
+    name = "uadklib",
+    autogen = True,
+    configure_command = "conf.sh",
+    configure_in_place = True,
+    configure_options = [
+        "--enable-static",
```

Comment:
one other thing im just thinking - are we building the absolute minimum required?

no reason to think otherwise - just that historically we have had issues with foreign_cc builds building a lot of unnecessary code